airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/query.py

```diff
@@ -1984,3 +1984,174 @@ def record_process_components(self, record: MutableMapping[str, Any]) -> Iterabl
+        Field(name="appStoreDeveloperUrl", alias="app_store_developer_url"),
+        Field(name="isPostPurchaseAppInUse", alias="is_post_purchase_app_in_use"),
+        Field(name="previouslyInstalled", alias="previously_installed"),
+        Field(name="pricingDetailsSummary", alias="pricing_details_summary"),
+        Field(name="pricingDetails", alias="pricing_details"),
+        Field(name="privacyPolicyUrl", alias="privacy_policy_url"),
+        Field(name="publicCategory", alias="public_category"),
+        Field(name="uninstallMessage", alias="uninstall_message"),
+        Field(name="webhookApiVersion", alias="webhook_api_version"),
+        Field(name="shopifyDeveloped", alias="shopify_developed"),
+        Field(name="id", alias="provider_id"),
+        Field(name="failedRequirements", alias="failed_requirements", fields=failed_reqirements_fields),
+        Field(name="feedback", fields=feedback_fields),
+    ]
+
+    assessments_fields: List[Field] = [
+        Field(name="riskLevel", alias="risk_level"),
+        Field(name="facts", fields=["description", "sentiment"]),
+        Field(name="provider", fields=provider_fields),
+    ]
+
+    risk_fields: List[Field] = [
+        "recommendation",
+        Field(name="assessments", fields=assessments_fields),
+    ]
+
+    # main query
+    query_nodes: List[Field] = [
+        "__typename",
+        "updatedAt",
+        Field(name="id", alias="order_id"),
+        Field(name="risk", fields=risk_fields),
+    ]
+
+    record_composition = {
+        "new_record": "Order",
+        # there are no record components provided for this stream.
+    }
+
+    def _process_assessments(self, assessments: Iterable[MutableMapping[str, Any]]) -> Iterable[MutableMapping[str, Any]]:
+        for assessment in assessments:
+            provider = assessment.get("provider", {})
+            if provider:
+                # save and resolve provider id
+                provider["admin_graphql_api_id"] = provider.get("provider_id")
+                provider["provider_id"] = self.tools.resolve_str_id(provider.get("provider_id"))
+        return assessments
+
+    def _has_risk_recommendation(self, recommendation: Optional[str]) -> bool:
+        # if there are no risk recommendation, the value is literally "NONE",
+        # we should skip such record, because there is no risk info for it.
+        no_risk_pattern = "NONE"
+        return recommendation != no_risk_pattern if recommendation else False
+
+    def record_process_components(self, record: MutableMapping[str, Any]) -> Optional[Iterable[MutableMapping[str, Any]]]:
+        """
+        Defines how to process collected components.
+        """
+        # unnest mandatory fields from their placeholders
+        risk = record.get("risk", {})
+        recommendation = risk.get("recommendation") if risk else None
+        # process records which has some risk recommendation
+        if self._has_risk_recommendation(recommendation):
```

Comment:
Do we have tests for that?

Reply:
Not dedicated one to cover the method specifically, but these transformations are covered within the `unit_tests/graphql_bulk/test_job.py` in `test_bulk_stream_parse_response`; the RAW BULK input and RECORD output are tested there.